src/EStimLibrary/Extensions/SpatialModel/StringHierarchy/StringHierarchySpec.cs

```diff
@@ -81,118 +117,70 @@ public static string JoinFullSpec(string[] regionSet, string[] modifierSet)
+        var nameElements = optionedRegionName
+            .Split(OPTION_REGION_DELIMITER, 
+                StringSplitOptions.RemoveEmptyEntries)
+            .Select(e => e.Trim())
+            .ToArray();
 
-    public bool ModifiersAllowOverlap(StringHierarchySpec other,
-        out string[] commonModifiers)
-    {
-        // Overlaps if all modifiers in shorter set are in longer set.
-        string[] lessSpecificModSet; // Temp variables to store mod set references.
-        string[] moreSpecificModSet;
-        if (this.ModifierSet.Length < other.ModifierSet.Length)
-        {
-            lessSpecificModSet = this.ModifierSet;
-            moreSpecificModSet = other.ModifierSet;
-        }
-        else
+        // Fail if more than 2 elements are present
+        if (nameElements.Length > 2)
         {
-            lessSpecificModSet = other.ModifierSet;
-            moreSpecificModSet = this.ModifierSet;
+            baseName = option = "";
+            return false;
         }
-        commonModifiers = lessSpecificModSet.Intersect(moreSpecificModSet)
-            .ToArray();
-        //bool modifiersOverlap = shorterModSet.All(m => longerModSet.Contains(m));
-        return commonModifiers.Length == lessSpecificModSet.Length;
-    }
 
-    public bool TryGetOverlap(StringHierarchySpec other,
-        out string overlappingRegion, out bool contains)
-    {
-        // Check if the region specs actually have a common sequence.
-        bool RegionSetsOverlap = this.RegionSetOverlaps(other,
-            out var overlappingRegionSet);
-        // This contains other if all this's elements in other.
-        bool containsRegionSet = RegionSetsOverlap &&
-            overlappingRegionSet.Length == this.RegionSet.Length;
-
-        // Overlaps if all modifiers in shorter set are in longer set.
-        bool modifiersOverlap = this.ModifiersAllowOverlap(other,
-            out var commonModifiers);
-        // Contains if all this's modifiers shared.
-        bool contiansModifiers = commonModifiers.Length ==
-            this.ModifierSet.Length;
-
-        // Determine overall overlap and containment.
-        bool overlaps = RegionSetsOverlap && modifiersOverlap;
-        contains = containsRegionSet && contiansModifiers;
-
-        // Determine overlapping region: take most specific region and mods.
-        overlappingRegion = overlaps ?
-            // Region spec: other if contained (more specific), else this.
-            JoinFullSpec(containsRegionSet ? other.RegionSet : this.RegionSet,
-                this.ModifierSet.Union(other.ModifierSet).ToArray())
-            : string.Empty;     // Empty if no overlap.
-
-        return overlaps;
+        // Process one or two elements
+        baseName = nameElements[^1].Trim();
+        option = nameElements.Length == 2 ? nameElements[0].Trim() : "";
+        return true;
     }
 
     public virtual bool Equals(StringHierarchySpec? other)
```

Comment:
Check the coverage report for Equals() and GetHashCode(). I think just a few test cases missing to do with when "other" is null.